api/models/people.py

```diff
@@ -135,6 +138,43 @@ def __str__(self) -> str:
         return f"{self.personal_info.full_name}. Status: {getattr(self, 'status')}"
 
 
+# TODO the module is getting big. We can move Coordinator... etc into separate modules,
+#  which will also allow to rename all model modules into singular (because "people.py"
+#  will only contain `Person`).
+class CoordinatorQuerySet(models.QuerySet["Coordinator"]):
+    def annotate_with_count(self) -> "CoordinatorQuerySet":
+        return self.annotate(group_count=Count("groups"))
+
+    def below_threshold(self) -> "CoordinatorQuerySet":
+        """QuerySet with coordinators with not enough groups."""
+        return self.annotate_with_count().filter(group_count__lt=CoordinatorGroupLimit.MIN)
+
+    def ok(self) -> "CoordinatorQuerySet":
+        """QuerySet with coordinators that are above threshold and within limit."""
+        return self.annotate_with_count().filter(
+            group_count__gte=CoordinatorGroupLimit.MIN,
+            group_count__lt=CoordinatorGroupLimit.MAX,
+        )
+
+    def limit_reached(self) -> "CoordinatorQuerySet":
+        """QuerySet with coordinators that have exceeded the limit of groups."""
+        return self.annotate_with_count().filter(group_count__gte=CoordinatorGroupLimit.MAX)
+
+
+class CoordinatorManager(models.Manager["Coordinator"]):
+    def get_queryset(self) -> CoordinatorQuerySet:
+        return CoordinatorQuerySet(self.model, using=self._db)
+
+    def below_threshold(self) -> CoordinatorQuerySet:
```

Comment:
Afaik, you are able to do smth like


```
class SomeModel:
    objects = SomeModelQuerySet.as_manager()
```

And then you don't need to duplicate these methods

Reply:
Same for all models